src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -106,6 +107,70 @@ public void writeAnnotation_withDescription() {
     assertEquals(writerVisitor.write(), "@SuppressWarnings(\"all\")\n");
   }
 
+  @Test
+  public void writeNewObjectExpr_basic() {
+    // isGeneric() is true, but generics() is empty.
+    ConcreteReference ref = ConcreteReference.withClazz(List.class);
+    TypeNode type = TypeNode.withReference(ref);
+    NewObjectExpr newObjectExpr = NewObjectExpr.builder().setIsGeneric(true).setType(type).build();
+    newObjectExpr.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), "new List<>()");
+  }
+
+  @Test
+  public void writeNewObjectExpr_withMethodExprArgs() {
+    // isGeneric() is false, and generics() is empty.
+    // [Constructing] `new IOException(message, cause())` where `cause()` is a method invocation.
+    TypeNode type = TypeNode.withReference(ConcreteReference.withClazz(IOException.class));
+    Variable message = Variable.builder().setName("message").setType(TypeNode.STRING).build();
+    VariableExpr msgExpr = VariableExpr.builder().setVariable(message).build();
+    MethodInvocationExpr causeExpr =
+        MethodInvocationExpr.builder()
+            .setMethodName("cause")
+            .setReturnType(TypeNode.withReference(ConcreteReference.withClazz(Throwable.class)))
+            .build();
+    NewObjectExpr newObjectExpr =
+        NewObjectExpr.builder()
+            .setType(type)
+            .setArguments(Arrays.asList(msgExpr, causeExpr))
+            .build();
+    newObjectExpr.accept(writerVisitor);
+    assertEquals(writerVisitor.write(), "new IOException(message, cause())");
+  }
+
+  @Test
+  public void writeNewObjectExpr_withGenericsAndArgs() {
+    // isGeneric() is true and generics() is not empty.
+    ConcreteReference listRef =
+        ConcreteReference.builder()
+            .setClazz(List.class)
+            .setGenerics(Arrays.asList(ConcreteReference.withClazz(Integer.class)))
+            .build();
+    ConcreteReference mapRef =
+        ConcreteReference.builder()
+            .setClazz(HashMap.class)
+            .setGenerics(Arrays.asList(ConcreteReference.withClazz(String.class), listRef))
+            .build();
+    TypeNode type = TypeNode.withReference(mapRef);
+    MethodInvocationExpr methodExpr =
+        MethodInvocationExpr.builder()
+            .setMethodName("foobar")
+            .setReturnType(TypeNode.INT)
+            .setStaticReferenceName("SomeClass")
```

Comment:
This will need to be merged with the recent staticRefType change in PR #122.

Reply:
sure, approved the PR and will make a change after it's merged.

Reply:
done.